### ISSUE TITLE
Suggestion: Generate bad mesh results in Rhino w. Butterfly

### ISSUE THREAD

**Issue by MikkiSeidenschnur** [ladybug-tools/butterfly#216]
Hi, just a small wish for a component that could be implemented in the future:
- I would like to be able to see the "Bad mesh" cells that is generated during the "checkMesh" I've noticed that some files are written, for instance if there are non-orthogonal faces, checkMesh will give you a set called "nonOrthoFaces". I know that this can be visualized in Paraview, just think that it would be nice to be able to check this in rhino/grasshopper and save the time spend transferring the case to paraview, and then going back to Butterfly to make a new mesh. 

I would have a go at it myself if I had the time, but I'm thinking that it's fairly easy to implement. 

Best regards, and keep up the good work,
Mikki Seidenschnur

**Comment by mostaphaRoudsari** [comment]
Good one! I would say, it should be one of the default outputs of `checkMesh`.
Do you have a case that I can generate such a mesh? We just need to know where the file is and what is the structure of the mesh in that file.

**Comment by MikkiSeidenschnur** [comment]
Yeah, I have one with "nonOrthoFaces". It is in the 3.org folder in "sets". Actually now that I think about it, I don't think that paraview in windows can visualize these - therefore it would be even cooler to get it implemented. 

Best regards
Mikki

**Comment by mostaphaRoudsari** [comment]
Thanks for sharing. I leave a note here for myself. The file includes the face numbers that should be extracted from the mesh file which then means extracting the points from the point file. The challenge is to make the process memory-efficient.

```c++
/*--------------------------------*- C++ -*----------------------------------*\
| =========                 |                                                 |
| \\      /  F ield         | OpenFOAM: The Open Source CFD Toolbox           |
|  \\    /   O peration     | Version:  v1612+                                |
|   \\  /    A nd           | Web:      www.OpenFOAM.com                      |
|    \\/     M anipulation  |                                                 |
\*---------------------------------------------------------------------------*/
FoamFile
{
    version     2.0;
    format      ascii;
    class       faceSet;
    location    "2/polyMesh/sets";
    object      nonOrthoFaces;
}
// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //


2688
(
1609728 
1609729 
1605657 
1605658 
1609755 
1605659 
1609756 
1605660 
1609757 
1609758 
1605663 
1605664 
1609761 
1605665 
1609762 
1605666 
.
.
.
```

**Comment by TheodoreGalanos** [comment]
@MikkiSeidenschnur You can visualize the checkMesh 'error' cells by ticking the use cellSet option in paraFoam.

This would be nice if we can add in Rhino indeed as it shows potential areas of mesh refinement.

Kind regards,
Theodore.

**Comment by MikkiSeidenschnur** [comment]
@TheodoreGalanos I actually don't see the option when running it in Paraview for windows. When I'm using paraview on Ubuntu I use that option all the time. Don't know if I'm having a weird version of Paraview or something. 

Best regards
Mikki

**Comment by TheodoreGalanos** [comment]
@MikkiSeidenschnur I think it might be a version issue. Not sure though.

**Comment by MikkiSeidenschnur** [comment]
Normally this is where I find it. This is a screenshot from windows:
![ms_setsnonorthofaces](https://cloud.githubusercontent.com/assets/20814326/24709573/a230275c-1a1a-11e7-94c3-4acbe6a06a12.PNG)

**Comment by TheodoreGalanos** [comment]
@MikkiSeidenschnur Yes it should be somewhere there. This is where I see it on my Linux paraFoam (v5.0.1)

![parafoam](https://cloud.githubusercontent.com/assets/13436572/24710027/6cf8f846-1a4e-11e7-8ada-8679c9c43fb1.png)

**Comment by TheodoreGalanos** [comment]
@MikkiSeidenschnur Yes this is where it should be. This is where I see it in Linux paraFoam (v5.0.1)

![parafoam](https://cloud.githubusercontent.com/assets/13436572/24710077/95429302-1a4e-11e7-9618-a23dc0e3b117.png)

**Comment by MikkiSeidenschnur** [comment]
@TheodoreGalanos Just to follow up on this: 
Windows Paraview does not have the feature to include the "Sets". Which is also way I would say it would be cool to get the feature in Butterfly.
